src/include/test/test_utils.h

```diff
@@ -37,24 +37,34 @@
 
 template <std::ranges::range R>
 void randomize(R& r, std::tuple<int, int> range = {0, 128}) {
+  using element_type = std::ranges::range_value_t<R>;
+
   std::random_device rd;
   // std::mt19937 gen(rd());
   std::mt19937 gen(2514908090);
 
-  if constexpr (std::is_floating_point_v<std::ranges::range_value_t<R>>) {
-    std::uniform_real_distribution<std::ranges::range_value_t<R>> dist(
+  if constexpr (std::is_floating_point_v<element_type>) {
+    std::uniform_real_distribution<element_type> dist(
         std::get<0>(range), std::get<1>(range));
     for (auto& x : r) {
       x = dist(gen);
     }
-    return;
   } else {
-    std::uniform_int_distribution<std::ranges::range_value_t<R>> dist(
-        std::get<0>(range), std::get<1>(range));
-    for (auto& x : r) {
-      x = dist(gen);
+    if constexpr (sizeof(element_type) == 1u) {
```

Comment:
I would suggest using `uint16_t` or `int16_t` so that you can be sure of the size.  I am also wondering if `%` is better than `/`.  (I don't have an answer for sure -- for some prngs `/` may be better statistically -- but that probably isn't the case for modern libraries.)  But using % would also let you guarantee that you have an 8 bit thing at the end.  Or you could just mask off 8 bits.

Reply:
I agree with using `uint16_t` instead of `short`. Regarding the second part of your commend, using `/` or `%`. The modulo operator should work fine, but in that case we can simply just downcast the value right? (modulo 256 is the same as casting to `uint8_t`). Do you want me to change it? :)